Azure/communication-ui-library-android
# [Calling][Feature] User facing diagnostics

## Purpose
This PR introduces User Facing Diagnostics notifications on the UI SDK.
User Facing Diagnostics are enabled by default and will show 2 types of notifications for Call Diagnostics during a call:

- A Top Message Bar that will be persistent unless the diagnostic is no longer relevant or the user dismisses the notification pressing the X button on the right.
- A type of Toast notification that appears on the bottom of the screen with information that will disappear after 4 seconds and cannot be dismissed by the user.
* ...

## Does this introduce a breaking change?


- [X ] No

## Pull Request Type
What kind of change does this Pull Request introduce?


- [ X] Feature

## Pull Request Checklist



- [X ] Synced with iOS, Web team
  - [X ] Screen captures included
  - [X ] Tested for Light/Dark mode
  - [ X] Tested for screen rotation
  - [ X] Tested on Tablet and small screen device (5")
  - [ X] Include localization changes
  - [X ] Tested on API 21, 26 and latest 
  - [ X] Tested for background mode
  - [ X] Unit Tests Included
  - [ X] UI Automated Tests Included

## How to Test

* Join a ACS group call or a Teams meeting.
* Once the call is connected, mute the microphone.
* Start speaking.
* See a notification on the bottom of the screen that says "You are muted"


## What to Check
Verify that the following are valid
* ...

## Other Information
